back/app/services/project_copy_service.rb

```diff
@@ -73,6 +76,54 @@ def export(
 
   private
 
+  def yml_content_builder_layouts(shift_timestamps: 0)
+    layout_images_mapping = {}
+
+    layouts = ContentBuilder::Layout.where(content_buildable_id: @project.id).map do |layout|
+      craftjs = layout.craftjs_jsonmultiloc
+
+      craftjs.each_key do |locale|
+        craftjs[locale].each_key do |node|
+          next unless craftjs[locale][node]['type']['resolvedName'] == 'Image'
+
+          source_image_code = craftjs[locale][node]['props']['dataCode']
+          new_image_code = SecureRandom.uuid
```

Comment:
We could try to keep the logic of generating the code in one place. E.g., in a static method `ContentBuilder::LayoutImage.generate_code`. It can be used here and in the instance method `generate_code` like `self.code = self.class.generate_code`.

Reply:
Not sure I follow.

We need to generate the uuid here, so we can map it to the old uuid and pass the mapping to the method that creates the yml entries for the images.

If we use a method in the image model, we won't know what the UUID will be when we create the mapping of UUIDs here, which is why I override the code generation in the model:
```
 def generate_code
      self.code = SecureRandom.uuid unless code
    end
 ```

Reply:
You will also use it here
```ruby
source_image_code = craftjs[locale][node]['props']['dataCode']
new_image_code = ContentBuilder::LayoutImage.generate_code
```